File: ruby/raindrops/raindrops.rb
class Raindrops

  def self.convert(num)
    output = NOISES.map { |factor, noise| noise if num % factor == 0 }
    # trick: nils (from above) are falsey, and "arr.any?" means
    # "arr.any? { |elt| elt.truthy? }", NOT "arr.size > 0"!
    return output.any? ? output.join : num.to_s
  end

  private

  NOISES = { 3 => "Pling", 5 => "Plang", 7 => "Plong" }

end

File: python/raindrops/raindrops.py
NOISES = ((3, "Pling"), (5, "Plang"), (7, "Plong"))

def convert(number):
    if not isinstance(number, int): raise ValueError
    result = [noise for trigger, noise in NOISES if number % trigger == 0]
    return "".join(result) if result else str(number)  # empty list is falsy
